Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift

```diff
@@ -175,12 +193,124 @@ struct UserDetailsView: View {
+private extension View {
+    @ViewBuilder
+    func deleteUser(in view: UserDetailsView) -> some View {
+        sheet(
+            isPresented: view.$presentUserPicker,
+            onDismiss: {
+                view.presentUserPicker = false
+            },
+            content: {
+                pickAnotherUser(in: view)
+            }
+        )
+        .alert(
+            UserDetailsView.Strings.deleteUserConfirmationTitle,
+            isPresented: view.$presentDeleteConfirmation,
+            presenting: view.selectedUser,
+            actions: { attribution in
+                Button(role: .destructive) {
+                    Task {
+                        view.deleteUserViewModel.didTapDeleteUser {
+                            view.dismissAction()
+                        }
+                    }
+                } label: {
+                    Text(UserDetailsView.Strings.deleteUserConfirmButtonTitle)
+                }
+            },
+            message: {
+                Text(UserDetailsView.Strings.deleteUserConfirmationMessage(username: $0.displayName))
+            }
+        )
+        .alert(
+            "Error",
+            isPresented: view.$presentDeleteUserError,
+            presenting: view.deleteUserViewModel.error,
+            actions: { _ in
+                Button("OK") {
+                    view.presentDeleteUserError = false
+                }
+            },
+            message: { error in
+                Text("An error occurred while deleting the user.")
+                // TODO: Use appropriate localized error message
+                Text(error.localizedDescription)
+            })
+    }
+
+    @ViewBuilder
+    func pickAnotherUser(in view: UserDetailsView) -> some View {
+        NavigationView {
+            Form {
+                VStack(alignment: .leading) {
+                    Text("You have specified this user for deletion:")
+                    Text("ID #\(view.user.id): \(view.user.username)")
+                }
+                .frame(maxWidth: .infinity)
+                .listRowBackground(Color.clear)
+                .listRowInsets(.zero)
+
+                Section {
+                    if view.deleteUserViewModel.isFetchingOtherUsers {
+                        LabeledContent("Attribute all content to:") {
+                            ProgressView()
+                        }
+                    } else {
+                        Picker("Attribute all content to:", selection: view.$selectedUser) {
+                            ForEach(view.deleteUserViewModel.otherUsers) { user in
+                                Text("\(user.displayName) (\(user.username))").tag(user)
+                            }
+                        }
+                    }
+                }
+            }
+            .navigationTitle("Delete Confirmation")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
```

Comment:
The navigation bar button should be black in jpios. For `UINavigationController`, it's handled using the appearance. We need something similar for `NavigationView` that gets applied globally.

Reply:
Good catch! I tried a few things and couldn't get it to work. Do you happen to know a nice way to apply a tint color on all navigation bar buttons. `tint()` would work obviously, but I want to avoid having to setting `tint` on every navigation bar button.

I checked this sheet SwiftUI view at runtime. It uses a `UINavigationBar` subclass, which should just work: it should use the tint color in the `UINavigationBar.appearance()` set in the app code. But somehow it doesn't reuse that.

Reply:
I don't from the top of my head. This is partially why I've been using `UINavigationController` even for SwiftUI views.